R/get_st_data_path.R

```diff
@@ -0,0 +1,25 @@
+#' Get the path to the required stress test auxiliary data location. Can be
+#' provided either via environment variable or options.
+#'
+#' @param envvar String. Provide the name of the environment variable that points
+#'   to the data directory/repo. Default is "ST_DATA_PATH".
+#' @param option String. If no environment variable is provided and the default
+#'   "ST_DATA_PATH" cannot be found, the function tries to get the data path
+#'   from options. In that case, provide the name of the option that points
+#'   to the data directory/repo. Default option to get the path is "st_data_path".
+#'
+#' @family miscellaneous utility functions
+#'
+#' @return Character
+#'
+#' @export
+get_st_data_path <- function(envvar = "ST_DATA_PATH",
+                             option = "st_data_path") {
+  if (Sys.getenv(envvar) != "") {
+    return(fs::path(Sys.getenv(envvar)))
+  } else if (!is.null(options(option)[[option]])) {
+    return(fs::path(options(option)[[option]]))
+  } else {
+    stop("Please add data path as envvar or R option")
+  }
+}
```

Comment:
I was already happy I managed to write a test using withr, but I'll admit this is a lot nicer and the test much more thorough. Will implement

Reply:
I am changing the second and third tests a bit though, because it can have check() fail locally if the environment variable is set in the .Rprofile
I am doing the following to circumvent the issue and ensure that no environment variable gets matched:

```
test_that("works with `var` set as an R option", {
  path <- "/as/options"
  withr::local_envvar(list(ST_DATA_PATH = NULL))
  withr::local_options(list(ST_DATA_PATH = path))

  expect_equal(get_st_data_path(), path)
})
#> Test passed 😀

test_that("if `var` is unset errors gracefully", {
  withr::local_envvar(list(ST_DATA_PATH = NULL))
  expect_error(get_st_data_path(), "var.*unset")
})
#> Test passed 🎉
```

@maurolepore